### ISSUE TITLE
Problem with indexing

### ISSUE THREAD

**Issue by kyleellefsen** [rbnvrw/nd2reader#2]
I'm developing an image processing program - [flika](https://github.com/flika-org/flika) - that relies on this library. Thank you for maintaining (and thanks to @jimrybarski for writing it). 

I recently upgraded to a newer version of this reader and I'm having some trouble opening files. I've pinpointed the problem to the ```_calculate_image_group_number()``` method in the parser class in parser.py.  I can send you a sample file, it's 30 MB, bigger than I can attach here. The image is a z stack with 28 frames in total. When trying to access the second frame (index = 1) , this method returns an image group number of 28. Here is the function.

```python
def _calculate_image_group_number(self, frame_number, fov, z_level):
    """
    Images are grouped together if they share the same time index, field of view, and z-level.

    Args:
        frame_number: the time index
        fov: the field of view number
        z_level: the z level number

    Returns:
        int: the image group number

    """
    return frame_number * len(self.metadata["fields_of_view"]) * len(self.metadata["z_levels"]) + (
        fov * len(self.metadata["z_levels"]) + z_level)
```

I believe this can be fixed if ``len(self.metadata["z_levels"])`` is removed. I would create a pull request but I wanted to check if that simple solution might cause problems somewhere else. 

If you would like the .nd2 file, let me know where I can send it.

**Comment by rbnvrw** [comment]
Thank you for your bug report. Could you send the .nd2 file via WeTransfer to rbnvrw@gmail.com ?
Then I can check if removing `len(self.metadata["z_levels"])` would give problems.

**Comment by rbnvrw** [comment]
Thank you for sending the files, I will try to reproduce the issue and report back with a solution.

**Comment by kyleellefsen** [comment]
Great, thanks! I look forward to it.

**Comment by rbnvrw** [comment]
@kyleellefsen it is fixed for the three files you provided, see commit 26138af above.
Code example:
```python
with ND2Reader(r"trial_1.nd2") as nd2:
    nd2.iter_axes = 'z'
    self.assertEqual(len(nd2.metadata['z_levels']), 28)
    for z in nd2.metadata['z_levels']:
        self.assertEqual(nd2[z].shape, (1024, 1024))
```
The `nd2.iter_axes = 'z'` makes the reader iterate over the `z` axis instead of the `t` axis. `nd2.metadata['z_levels']` is an array of all `z` levels.

It would be great if you could test it with the code from the master branch, otherwise it will be fixed in the next release.

**Comment by kyleellefsen** [comment]
That fixed it. The following code works:

```python
>>> import nd2reader
>>> nd2 = nd2reader.ND2Reader(r'trial_1.nd2')
>>> nd2.iter_axes = 'z'
>>> nd2[1]
Frame([[ 416,  399,  396, ..., 7385, 7593, 7879],
       [ 408,  431,  446, ..., 8093, 7859, 8492],
       [ 427,  402,  424, ..., 8152, 8296, 8451],
       ...,
       [ 398,  409,  419, ..., 1445, 1569, 1546],
       [ 413,  425,  413, ..., 1464, 1532, 1515],
       [ 428,  426,  419, ..., 1519, 1503, 1521]], dtype=uint16)
```
However, if I don't set the iter_axes to z, I get an error. I see in the reader.py line 112, ``set.iter_axes = 't'``. For most users, most of the time, an nd2 file is only going to have multiple entries along _either_ the 't' dimension or the 'x' dimension, but not both. Because of this, I think it would be a good idea to check which axes should be the default by looking at the length of the axis. Something like this:
```python
self.iter_axes = 't'
if len(t_axis) == 1:
    if len(z_axis) == 1:
        self.iter_axes = 'c'
    elif len(z_axis) > 1:
        self.iter_axes = z
```

**Comment by rbnvrw** [comment]
This is a nice suggestion. It requires a bit more work to make it a bit more robust, but I will implement it for version 3.0.2. Thank you!

**Comment by kyleellefsen** [comment]
Thanks! And thanks for supporting this project, it is very useful!

**Comment by rbnvrw** [comment]
Solved in release 3.0.2
